packages/react-native/package.json

```diff
@@ -98,6 +98,7 @@
     "featureflags": "node ./scripts/featureflags/index.js"
   },
   "peerDependencies": {
+    "@react-native-community/cli-server-api": "*",
```

Comment:
@tido64 @robhogan 
mentioned that this will force `cli-server-api` to be installed (or at least result in a warning) also for expo users.

Can't we add this as direct dependency inside the template instead?

Reply:
We still need it here for it to be hoisted correctly. But we can make it optional and make it a direct dependency in the template.

Reply:
Basically, we declare that `@react-native/community-cli-plugin` depends on `cli-server-api` but that consumers need to satisfy this requirement. The only consumer of `@react-native/community-cli-plugin` is `react-native`, but you don't want `react-native` to depend on CLI any longer. So `react-native` needs to "forward" the dependency to its consumers, hence redeclare peer dependency. If this does not happen, package managers won't correctly link the package.

However, since `react-native` has `cli-server-api` as peer dependency, it's probably fine to mark it as optional and add it in the template as a direct dependency to satisfy the requirement.

Reply:
https://github.com/react-native-community/template/pull/105

Reply:
> but you don't want react-native to depend on CLI any longer.

I think there is some cleanup to make sure `react-native` doesn't need to depend on `@react-native/community-cli-plugin` at all so this change wont' be required at all

Reply:
For the time being, can't we just have:
1. Remove the `peerDependency` here between `react-native` and `@react-native-community/cli-server-api` (so amend this line)
2. Add the dependency on the template as per react-native-community/template#105

?

Reply:
I've replied to this offline but I'll post it here as well for transparency: The short answer is, if you do it this way, package managers won't put `@react-native-community/cli-server-api` in a place where `@react-native/community-cli-plugin` will be able to find it.

Longer answer: See https://github.com/facebook/react-native/pull/49325#discussion_r1953092387